### PR TITLE
Inventory & Assembly proposal

### PR DESCRIPTION
**Pull request by doomspork**
This isn't complete, I'll expand on how these will be used later today :+1:

### CODE REVIEW

File: protos/bottle/inventory/events/v1/component_availability_requested.proto [hunk 1]
```diff
@@ -0,0 +1,9 @@
+syntax = "proto3";
+
+import "bottle/inventory/v1/component.proto";
+
+package bottle.inventory.events.v1;
+
+message ComponentAvailabilityRequested {
```
Comment (btkostner) [inline protos/bottle/inventory/events/v1/component_availability_requested.proto:7]: Having a quantity request and change in queue seems weird to me, but I guess this makes sense with our current infrastructure. I know that aws elb support grpc if we want to start working with that for this project.

File: protos/bottle/assembly/v1/build.proto [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  BuildStatus status = 2;
+
+  message BuildComponent {
+    bottle.inventory.v1.Component component = 1;
+    int32 quantity = 2;
+  }
+
+  repeated BuildComponent build_component = 3;
```
Comment (btkostner) [inline protos/bottle/assembly/v1/build.proto:24]:
```suggestion
  repeated BuildComponent build_components = 3;
```

because it's repeated

**Review by btkostner — APPROVED** [review]
I was thinking about how the split here is at `BuildCreated` and not `OrderCreated` and at first I was against it long term. But thinking about it, long term we would want to split out `Fulfillment.Product` to be able to keep track of each quantity independently for returns anyway. So this split makes more sense long term than what I was thinking.

File: protos/bottle/inventory/events/v1/component_availability_requested.proto (hunk 1 above)
Comment (doomspork) [reply protos/bottle/inventory/events/v1/component_availability_requested.proto:7]: This has felt meh to me as well. My thinking was they don't need to be synchronous so it didn't seem necessary to introduce a mechanism to achieve that when it isn't required.

(We discussed in Slack, we'll use this for now and circle back to this later)

**Review by btkostner — APPROVED** [review]

**Review by btkostner — APPROVED** [review]

**Comment by doomspork** [comment]
Thanks for the help @btkostner :tada:

**Pull request merged**